File: App/Components/Photo/Properties.cs
﻿using System;
using System.Collections.Generic;

namespace Websilk.Components.Properties
{
    public class Photo: ComponentProperties
    {
        public override int Width
        {
            get
            {
                return 450;
            }
        }

        public Photo(Core WebsilkCore, ComponentView c, string Section) : base(WebsilkCore, c, Section)
        {
            //file|hover-file|orig-width|orig-height|url|new-window|window-name|alt-text|speed (0 - 8)
            //use-bg|bg-size|bg-position|use-tile|use-parallax|parallax-speed|parallax-offset (9 - 15)
            if(Component.dataField == "")
            {
                Component.dataField = "|||||||||||||||";
            }

            string[] data = Component.dataField.Split('|');
            if(data.Length == 0) { return; }
            string photo = "/components/photo/icon.png";
            if(data[0] != "")
            {
                string[] paths = data[0].Split('/');
                if(paths.Length == 2)
                {
                    photo = "/content/websites/" + S.Page.websiteId + "/photos/" + paths[0] + "/" + "tiny" + paths[1];
                }
                else
                {
                    photo = "/content/websites/" + S.Page.websiteId + "/photos/" + "tiny" + data[0];
                }
            }

            //load properties into input fields via javascript
            string js =
                "$('#propsTxtLink').val('" + data[4] + "');" +
                "$('#propsLstTarget').val('" + (data[5] == "1" ? "_blank" : "") + "');" +
                "$('#propsTxtWindowName').val('" + data[6] + "');" +
                "$('#propsTxtAlt').val('" + data[7] + "');" +
                "S.editor.components.properties.current.load();";

            if(data[1] == "") { js += "$('.winProperties .remove-hover-photo').hide();"; }

            S.Page.RegisterJS("photoprops", js);
            
            Data["preview-photo"] = photo;
            Data["helpicon-hover"] = "";
            Data["helpicon-link"] = "";
            Data["helpicon-alt"] = "";
            Data["helpicon-window"] = "";
        }

    }
}

File: App/Components/Photo/properties.js
var props = S.editor.components.properties;
props.menu.clear();
props.menu.add('Photo', '.props-pages', 1);
//props.menu.add('Advanced', '.props-pages', 2);
props.menu.select('.props-pages', 1);


//set up current properties object functionality
S.editor.components.properties.current = {
    photo: {
        update: function (src) {
            var id = S.editor.components.properties.selected.id.substr(1);
            S.ajax.post('/api/Components/Photo/UpdatePhoto', { id: id, photo: src, type: 1 }, S.ajax.callback.inject);
        }
    },

    photoHover: {
        update: function (src) {
            var id = S.editor.components.properties.selected.id.substr(1);
            S.ajax.post('/api/Components/Photo/UpdatePhoto', { id: id, photo: src, type: 2 }, S.ajax.callback.inject);
        },

        remove: function () {
            if (confirm('Do you really want to remove the photo that is used when the mouse hovers over the component?') == true) {
                var id = S.editor.components.properties.selected.id.substr(1);
                $('.winProperties .remove-hover-photo').hide();
                S.ajax.post('/api/Components/Photo/RemoveHover', { id: id }, S.ajax.callback.inject);
            }
            
        }
    },

    changeTarget: function () {
        if ($('#propsLstTarget').val() == '_blank') {
            $('.winProperties .props-window-name').show();
        } else { $('.winProperties .props-window-name').hide(); }
    },

    load: function () {
        //executed when the properties window first loads
        this.changeTarget();
    },

    save: function () {
        //save component properties when the user presses the "apply changes" button
        var options = {
            id: S.editor.components.properties.selected.id.substr(1),
            url: $('#propsTxtLink').val(),
            alt: $('#propsTxtAlt').val(),
            target: $('#propsLstTarget').val(),
            win: $('#propsTxtWindowName').val()
        };
        S.ajax.post('/api/Components/Photo/SaveProperties', options, S.ajax.callback.inject);
    }
}